.github/workflows/release.yaml

```diff
@@ -1,39 +1,117 @@
-name: release
+# This code is borrowed from https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/release.yaml
+name: Create Release
 
 on:
   push:
-    # Sequence of patterns matched against refs/tags
-    tags:
-    - "v*"
+    branches:
+      - main
+    paths:
+      - 'releasenotes/*.md'
 
 permissions: {}
 
 jobs:
-  build:
-    name: tag release
+  push_release_tags:
+    permissions:
+      contents: write
     runs-on: ubuntu-latest
-
+    outputs:
+      release_tag: ${{ steps.release-version.outputs.release_version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # tag=v4.1.7
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@e9772d140489982e0e3704fea5ee93d536f1e275 # tag=v45.0.1
+      - name: Get release version
+        id: release-version
+        run: |
+            if [[ ${{ steps.changed-files.outputs.all_changed_files_count }} != 1 ]]; then
+              echo "1 release notes file should be changed to create a release tag, found ${{ steps.changed-files.outputs.all_changed_files_count }}"
+              exit 1
+            fi
+            for changed_file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+              export RELEASE_VERSION=$(echo "${changed_file}" | grep -oP '(?<=/)[^/]+(?=\.md)')
+              echo "RELEASE_VERSION=${RELEASE_VERSION}" >> ${GITHUB_ENV}
+              echo "RELEASE_VERSION=${RELEASE_VERSION}" >> ${GITHUB_OUTPUT}
+              if [[ "${RELEASE_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$ ]]; then
+                echo "Valid semver: ${RELEASE_VERSION}"
+              else
+                echo "Invalid semver: ${RELEASE_VERSION}"
+                exit 1
+              fi
+            done
+      - name: Determine the release branch to use
+        run: |
+            if [[ ${RELEASE_VERSION} =~ beta ]] || [[ ${RELEASE_VERSION} =~ alpha ]]; then
```

Comment:
This makes us commit to releasing where alpha and beta are tags on main, but RC and actual release tag has to be in release branch.

Do we REALLY want to commit to?

Reply:
I think this is fine, this is something that is followed in the Kubernetes release, and CAPI also followed them.
What are the potential drawbacks of this? 
Keep creating betas until we have the code freeze and release RCs afterward.

Reply:
It is just commitment we need to agree to. 

We have not been doing too many branching RCs in the past, and in general not many alpha/betas either. We then can't do RC releases from main, but must cut the branch and fork the work at that point. We know branching for minor is not trivial with out multi-repo setup, so I just want us to make informed decision that this piece of code will dictate that process going forward.

For example, pushing a branch here does not mean we can test it in anyway unlike CAPI in a single repository setup.

Reply:
As discussed offline, let's mark this
/hold
For agreement next week when people are back from kubecon etc